src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/ThreadSafeRandom.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+
+namespace Azure.Base.Http.Pipeline
+{
+    internal class ThreadSafeRandom: Random
```

Comment:
Why do we need this type? We just need an int. Isn't random.getnext thread safe?

Reply:
Random is not thread-safe.

https://docs.microsoft.com/en-us/dotnet/api/system.random?redirectedfrom=MSDN&view=netframework-4.7.2#the-systemrandom-class-and-thread-safety

Reply:
We could potentially consider the [RNGCryptoServiceProvider](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.rngcryptoserviceprovider?redirectedfrom=MSDN&view=netcore-2.2#thread-safety) here, as it is explicitly thread safe.   Though, it may be overkill and potentially slower.

Reply:
Crypto provider is way overkill for this, I think I'll change to having a lock for now and can go thread-local path in case we see high contention on that lock.

Reply:
Wow, I did not realize it can start returning zeros.